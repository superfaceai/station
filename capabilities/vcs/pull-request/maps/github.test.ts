import { SuperfaceTest } from '@superfaceai/testing-lib';

describe(`vcs/pull-request/github}`, () => {
  let superface: SuperfaceTest;

  beforeEach(() => {
    superface = new SuperfaceTest();
  });

  describe('PullRequest', () => {
    it('should perform successfully', async () => {
      await expect(
        superface.run({
          profile: 'vcs/pull-request',
          provider: 'github',
          useCase: 'PullRequest',
          input: {
            owner: 'superfaceai',
            repo: 'astexplorer',
            identifier: 3,
          },
        })
      ).resolves.toMatchSnapshot();
    });

    it('should map error', async () => {
      await expect(
        superface.run({
          profile: 'vcs/pull-request',
          provider: 'github',
          useCase: 'PullRequest',
          input: {
            owner: 'superfaceai',
            repo: 'astexplorer',
            identifier: 9999,
          },
        })
      ).resolves.toMatchSnapshot();
    });
  });
});
