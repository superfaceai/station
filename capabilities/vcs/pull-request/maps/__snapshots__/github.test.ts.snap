// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vcs/pull-request/github} PullRequest should map error 1`] = `
Object {
  "error": "HTTPError: Expected HTTP error
AST Path: definitions[0].statements[3].responseHandlers[0].statements[1]
Original Map Location: Line 20, column 7",
}
`;

exports[`vcs/pull-request/github} PullRequest should perform successfully 1`] = `
Object {
  "value": Object {
    "id": 567476468,
    "sha": "a8e318f2f5f14504a2f0da049d26cbc80d35fa9d",
    "title": "chore: Bump parser version",
    "url": "https://api.github.com/repos/superfaceai/astexplorer/pulls/3",
  },
}
`;
